pywhy_graphs/algorithms/tests/test_generic.py

```diff
@@ -348,3 +348,72 @@ def test_valid_mag():
     admg.add_edge("H", "J", admg.undirected_edge_name)
 
     assert not pywhy_graphs.valid_mag(admg)  # there is an undirected edge between H and J
+
+
+def test_dag_to_mag():
```

Comment:
You should add a test to each mag that it is not almost cyclic and also maximal perhaps even. The graphs are small, so the maximality check should not take long

Reply:
So I added a test to check that no MAG has an ADC. How do I check for maximality? I see the definition in the paper but could you roughly write down the algorithm?

Reply:
https://www.jmlr.org/papers/volume9/zhang08a/zhang08a.pdf states: "ii. there is no inducing path between any two non-adjacent vertices (maximal)."

Reply:
Ah alright.

Reply:
So I am not passing any L or S variables to the is_maximal function because an MAG cannot have latent or selection variables.